src/TriggerBinding/SqlTriggerMetricsProvider.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.Extensions.Logging;
+using static Microsoft.Azure.WebJobs.Extensions.Sql.SqlTriggerConstants;
+using static Microsoft.Azure.WebJobs.Extensions.Sql.SqlTriggerUtils;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Sql
+{
+    /// <summary>
+    /// Provider class for unprocessed changes metrics for SQL trigger scaling.
+    /// </summary>
+    internal class SqlTriggerMetricsProvider
+    {
+        private readonly string _connectionString;
+        private readonly ILogger _logger;
+        private readonly SqlObject _userTable;
+        private readonly string _userFunctionId;
+
+        public SqlTriggerMetricsProvider(string connectionString, ILogger logger, SqlObject userTable, string userFunctionId)
+        {
+            this._connectionString = !string.IsNullOrEmpty(connectionString) ? connectionString : throw new ArgumentNullException(nameof(connectionString));
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._userTable = userTable ?? throw new ArgumentNullException(nameof(userTable));
+            this._userFunctionId = !string.IsNullOrEmpty(userFunctionId) ? userFunctionId : throw new ArgumentNullException(nameof(userFunctionId));
+        }
+        public async Task<SqlTriggerMetrics> GetMetricsAsync()
+        {
+            return new SqlTriggerMetrics
+            {
+                UnprocessedChangeCount = await this.GetUnprocessedChangeCountAsync(),
+                Timestamp = DateTime.UtcNow,
+            };
+        }
+        private async Task<long> GetUnprocessedChangeCountAsync()
+        {
+            long unprocessedChangeCount = 0L;
+
+            try
+            {
+                long getUnprocessedChangesDurationMs = 0L;
+
+                using (var connection = new SqlConnection(this._connectionString))
+                {
+                    this._logger.LogDebugWithThreadId("BEGIN OpenGetUnprocessedChangesConnection");
```

Comment:
Where does the log go, kusto FunctionLogs table?  Do we want to log it on every call?

Reply:
This is logging at the debug level so I would only expect it to be logged if the log level is set to that by the customer. And information like this is useful for helping track down issues (for example if the connection hangs on opening this would help indicate that).

Reply:
yes, the AppInsights logs will no be published. I am concerning about  kusto FunctionLogs table as we log the all levels there. I am wondering about the amount of the logs as the checks will be executed periodically. We can log if an operation time was exceeded some expected threshold or an exception was occurred.

There lot of "LogDebugWithThreadId" introduced in the PR. Can you try to deploy a function app + new SQL extension with TBS + RDS enabled and see how much new logs are generated in FunctionLogs kusto table.